### DISCUSSION TITLE
2024菜鸟自学笔记 Data science & AI, Blockchain & Crypto

### DISCUSSION THREAD

**Discussion by Lewisrxliu** [RoadSideUniversity/RSU#6] [General]
2024.3.14
1)Advanced LLM AIgentic Frameworks
Sam Witteveen - CoFounder & CEO Red Dragon AI
Google Developer Expert Machine Learning & Deep Learning
sam@reddragon.ai
@sam_witteveen
2) LLM AGENTS IN B2C
USE-CASES AND EVALUATION
BY PRAVEEN GOVINDARAJ
3) https://bit.ly/MLSG_2024-03?r=qr
• DSPy Discord
• Full-Stack DSPy Application with FastAPl and Streamlit
• Gemma fine-tuning in Colab & Tokenizer 
RED DRAGON AI

**Comment by Lewisrxliu** [comment]
2024.3.15 
# The Gen AI Payoff in 2024, databricks-VP of gen AI, Mr Naveen Rao 
Enterprise case: Ford F-150, Key Concerns is Application quality - Accuracy, safety, governance
Challenges - Building and deploy product quality Gen AI solutions
Enterprise case :  Air Canada honor of their chatbot mistake and refund, no monitoring, content filters, safety, violate and governance, need enterprise customized solution, enterprise need: 1.1) product quality - accurate safe governed genAI apps, 1.2) complete control-complete ownership control and security of model and data, 1.3) lower cost
Databricks is leading platform for maximizing gen AI quality, best place to build customized gen AI cost effectively 
Start to build gen AI solution by: pre-training, fine tuning, RAG - Retrieval Augmented generation by combined LLM with enterprise data, Prompt engineering craft specialized prompt 
Enterprise case : Replit online integrated development built Ghostwriter- code gen model, 1-3 days, lower cost pretraining stable diffusion model now 10x less cost
# RAG - Retrieval Augmented generation by  Mr Eric Peter Snr 
RAG: Chain orchestratrator, embedding LLM, vector database return to Q, Prompt, getting high quality accurate RAG is hard - measure, evaluate, collect feedback, improve, guardrail, 
RAG Tools preview Demo: integrated ML flow and Mosaic AI: Chain.invoke, rag.confiig.get (llm-prompt-template, Azure-openai-
Log the chain to UC, Deploy the chain (model_name, version=model version),
Try: Ares - RCG, Source, Inspect logs, 
LLM Judges, scale human evaluate by using LLM, (Eval Set, jusdges, config=,), RAG studio Eval Dashboard, faithful on gen to given context
Try again: improved answer of ARES
# Testimonial ADP genAI stragety and partner with Databricks Mosaic AI by VP of ADP AI, Mr Fernando Schwartz
ADP providing tailored solutions support HR, Payroll, time management, benefit, talents, USA, 
Secure, ethical, responsible for easy payroll, smart analytics, human insights, search by intent serving HCM dataset
Challenges- how to achieve quality, protect IP, access unstructured data, control cost to achieve ROI
ADP data and AI strategy: Evaluate brand via leadership, improve business operation, reimaging product, commercialize data, OneAI central AI platform- common service managing AI model by Mosaic AI (model serving, vector search, data preparation, LLMOps, Finetuning, RAG studio), OneData central data platform by Delta lake unified data storage and Unity Catalog on governance and data management
# product showcase demo- Miss Nacy Hung, Mosaic AI engineer
Mosaic AI Fine tuning, instruction fine tuning or continue pretraining
Finetuning custom code HuggingFace, Demo txt to Seql, (ft SDK, monitoring finetuning run)
# Ms Lan Guan, Chief AI officer Accenture, partner testimonial
Conversational AI assitant change toward elevated human agent and proactive invisible service
reimagined genAI insight engine, spot check, meta data, keyword based, prototype LLM toward high throughput customized LLM
Acenture service reinvention with genAI, Evoling from call centre to genAI nerve center, solving customer problem, reduce complexity
# research showcase by Jonathan Frankle- chief scientist databricks, Matei Zaharia Cofounder CTO of databricks, UC Berkeley
data quality matters (FLOPS  10^24vs Core average error 0.4 CPT 70b), Llama 2-13B continued pretraining (Tokens 60B vs core average 41 by OWM), 
Compound AI system - a major shift by Matei, shift from 2023 model focus to 2024 focus on multiple compound AI system (Alphacode 2, AlphaGeometry, Gemini on MMLU, ChapGPT Plus), challenges - how to best optimize end-to-end system, how to evaluate each step quality, 
ARES - Automated RAG evaluation (Jon Saad Falcon, Omar Khattab, Christopher Potts, Matei Zaharia)
DSPy Auto-Optimizing AI system , compiling Declarative language model cal to self improve), dspy.AI
Next step : take free gen AI foundation course and get cert, sign for preview new RAG and FineTuning
DataAI summit, SanFan 2024

**Comment by Lewisrxliu** [comment]
2024.3.13 google developer space 
# Gemini based Duet AI embedded with VSCode
# enable Duet AI in Cloud Consol/cloud shell
Generate code/prompt design /prompt structure/Duet AI model
# Deploy...API implemented
https://cloud.google.com/ai/generative-ai?hl=zh_CN
https://codelabs.developers.google.com/codelabs/cloud-developer-duetai?hl=en#0
https://codelabs.developers.google.com/codelabs/duet-ai-workshop?hl=en#3
GDG workshop
-
Download the code and use Duet AI to learn more about Jump Start Solutions and the code you deployed.
1. If you have navigated away, click Cloud Code 
2. 
3. Click Cloud Functions
    * If you see a Select a project button, click it, and select the project you are working with.
    * Click the annotate-http function
    * Click the Download to new workspace

Calling the existing Cloud Function endpoint
Explore the code and call the endpoint you deployed in the Jump Start Solution.
1. Enter the prompt:
    * IMPORTANT Replace my-project-with-duet in the following prompt with your project id.
    * The region is us-central1 and the project id is my-project-with-duet. Write a curl post for the /annotate-http/annotate Cloud Function endpoint that performs label detection on the default eiffel tower image.
    * You should receive a response that includes a curl command like this:
......

**Comment by Lewisrxliu** [comment]
2024.3.19 Data Monetization with Looker @Google by Cloudmile/google CloudBI, 
Current data analytics: Global usage of BI <25%
#Datadriven model outperforming
LookML model looker data-warehouse SQL vs traditional MI by multi function
Buildin (google, AWS, Azure): in database architecture, Look ML semantic modelling, cloud native API
Demo by Francis Leung: Dashboard: business pulse, data driven alert, explore
Monetization turn data to revenue by Mr Brain Kuo, BI sales SME, Japac google cloud,Embedded analytics, 1)Diamond: Customization as a service, single tenancy, 2.Direct Monetization: Freemium (bronze, silver, gold)Features, scope, enhanced content, Killer features: Drilling for operational workflows, Explore,Alerts, custom actions, 3.Indirect Monetization: Proof of Value,Drive Behaviors & Engagement,Increase transactions
Retention, Competitive solution
Using Looker enable and drive new biz model: example Peloton # retention, engagement, proof of value, 
#Embedded analytic platform: insight in every product, cross devices omichannle 
Local options: OEM white labelled looker, embedded iframes display data visualisation and BI capacities, API using Looker model layer to display for highly customised UX
Demo: WIX using looker API for UX 
Demo: ecomm Atomfasion
Demo: Case by Lu Li, example Spire, WIX, Loreal, Carrefour, postman, Fox, Globalpayment, UiPath insight
#Looker& Gen AI: francis Leung, renewed spotlight, Looker + Vertex AI GenAI studio: accessible accurate data, fast &limitless insight, data privacy, Demo in Looker by Prompt

**Comment by Lewisrxliu** [comment]
2024. 4.10 课堂笔记 关于黄金和BTC的意见：
HSir关于黄金的指导意见： 黄金作为交易媒介稳定性的特点， 化学稳定性，成本稳定性，历史稳定性都为大众接受，但是黄金似乎已经接近完成人类社会的历史使命。
人类不可能的三角关系：认知，逻辑，控制
HSir关于BTC的指导意见：以透明度去中心化账本科技化低成本定时减半， 对抗物理暴力，对抗经济暴力，对抗政治暴力 = 未来世界的数字暴力， 未来不可能同时使用两种货币，BTC= first currency

**Reply by Lewisrxliu** [reply]
2024.4.17 课堂笔记 Hsir & XP 再次关于BTC的指导意见：
HSir关于BTC的指导意见： 观察矿工这一具有稳定收益模式的稳定群体对于后市的指导性意义，BTC矿工的出场与离场类似实体黄金矿场引领市场前驱的规律模式
#Plotting the future 方向与时间：追求确定性，未来不可知，经验不可靠，怎么办？HSir金句: #爱他，做时间的朋友！现在做功帮助收敛， Sam Altman - Backward Flat, forward vertical
Sam Altman 2017:
The hard part of standing on an exponential curve is: when you look backwards, it looks flat, and when you look forward, it looks vertical. And it’s very hard to calibrate how much you are moving because it always looks the same.
Sam Altman Dec 4, 2022:
interesting to me how many of the ChatGPT takes are either "this is AGI" (obviously not close, lol) or "this approach can't really go that much further".
#trust the exponential. flat looking backwards, vertical looking forwards.
XP分享观察：化工公司过去转型新能源-得到重新估值的机会，现在需要再寻找机会重新估值

**Comment by Lewisrxliu** [comment]
2024.5.28 Web3 Real world asset tokenised 
Adrien Delaroche Mr- Principal Architect web3 | crypto - Google
Google Blockchain tech offering
Node management 
Blockchain data BigQuery
Biggiest RWA Tokenisation is USDT followed by USDC
Challenges are regulation 
Trust-less ecosystem: USDC source code
If world need RWA then better do it in blockchain
RWA realtime data is critical : example of CitaDAO IRO of Singapore 18 SinMing Lane 0711 audited by ABDK
Web3 architecture Realtime data is needed beyond RWA : DEX , ERC-1400
How:
RPC node, Ethers.js, dApp (CiTADAO): problem is no scalable, RPC node is unable to respond million request, database need streaming optimisation -> AIchemy notify , moralis stream IRO> handle webhook> cloudSQL DB > In sync must be there - the MOST important thing
Build by Own vs Buy 
Chain reorg : Robust confirmation mechanism is essential
Must design architecture correctly (listener: invest
Every Web3 app need real time data (including RWA projects)
*   Don't connect your dApp to RPC nodes directly, use event driven architecture :) wor for web3 as well!
*   Build your own or use a provider, depending on your requirements
*   Cache your blockchain data
*  Decouple listening & business logic

Sherry Jiang: exGoogler, payment expert, founder CEO of Peek & BlueJay
Lesson from Google pay to Crypto
Google pay just focus on Peer to peer payment

Future RWA: open finance? Tokenise T-bill - Step One? - not for retail investor? Narrative protocol is RWA - Thousand project?

**Comment by Lewisrxliu** [comment]
Reference: channel 架构师之路![image](https://github.com/user-attachments/assets/a253299d-20ba-44ef-8181-270d2f0f86a2)
![image](https://github.com/user-attachments/assets/8d9024cc-ded7-4269-9eb1-a7f5bef72801)
![image](https://github.com/user-attachments/assets/58c238dd-f961-497f-a477-a3b501849f53)
![image](https://github.com/user-attachments/assets/822a08d1-6c24-49be-b467-43ecbb6717f5)
![image](https://github.com/user-attachments/assets/13eb09aa-9ad8-4559-8f38-dff4aaee9a52)
![image](https://github.com/user-attachments/assets/022f1c10-470a-452b-a0fd-1e58bd08bd3a)

**Comment by Lewisrxliu** [comment]
2024.8.27 ROS Robot Operation System

t.me/ROSMeetupSg
- [ ] Resources
ros.org
index.ros.org
discourse.ros.org
robotics.stackexchange.com
Open- RMF
osrf.github.io/ros2multirobotbook
github.com/osrf/rmf
_demos
github.com/osrf/rmf core
Open Source Robotics Foundation
•
gazebosim.org
ignitionrobotics.org
answers.gazebosim.org
community.gazebosim.org
- [ ] Agenda
* ﻿﻿RMF Site improvements with Rust and Bevy Reuben Thomas, Intrinsic Intern 
* ﻿﻿Experiences and lessons learned in scaling Open-RMF deployments Xiyu Oh, 
- [ ] Event Calendars
OSRF related events submission
bit.ly/OSRFCalendarForm
* ﻿﻿Open Source Robotics Foundation official events bit.ly/OSRFCalendar
* ﻿﻿Open Source Robotics Foundation community events bit.ly/OSRFCommunityCalendar

- [ ] Working with Rust and Bevy
1 Cross Platform
Runs natively on Windows, Mac, Linux and Browser
2 Extensibility
It beçomes
easy to extend
TunctionaLity by building new pl
3 Memory Safety
Rust follows a principle of "safety default”
4 Developer Experience

![image](https://github.com/user-attachments/assets/96d1358a-03df-4796-a640-32f07382e263)
![image](https://github.com/user-attachments/assets/c281695d-3d8a-480a-bb8e-de9485cdc93e)

**Comment by Lewisrxliu** [comment]
https://week.token2049.com
